### PR TITLE
Add FindPackage substitution

### PR DESCRIPTION
**Pull request by ivanpauno**
A convenient substitution for finding a package.
Nice to have for launch_frontend work.

### CODE REVIEW

File: launch_ros/launch_ros/substitutions/find_package.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+
+class FindPackage(Substitution):
+    """
+    Substitution that tries to locate the package prefix of a ROS package.
+
+    The ROS package is located using ament_index_python.
+
+    :raise: ament_index_python.packages.PackageNotFoundError when package is
+        not found during substitution
```
Comment (hidmic) [inline launch_ros/launch_ros/substitutions/find_package.py]: @ivanpauno consider:
```suggestion
        not found during substitution.
```

File: launch_ros/launch_ros/substitutions/find_package.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        """Getter for package."""
+        return self.__package
+
+    def describe(self) -> Text:
+        """Return a description of this substitution as a string."""
+        pkg_str = ' + '.join([sub.describe() for sub in self.package])
+        return 'Pkg(pkg={})'.format(pkg_str)
+
+    def perform(self, context: LaunchContext) -> Text:
+        """Perform the substitution by locating the executable."""
```
Comment (hidmic) [inline launch_ros/launch_ros/substitutions/find_package.py]: @ivanpauno that docstring needs an update.

File: launch_ros/launch_ros/substitutions/find_package.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    The ROS package is located using ament_index_python.
+
+    :raise: ament_index_python.packages.PackageNotFoundError when package is
+        not found during substitution
+    """
+
+    def __init__(self, package: SomeSubstitutionsType) -> None:
+        """Constructor."""
+        super().__init__()
+        self.__package = normalize_to_list_of_substitutions(package)
```
Comment (hidmic) [inline launch_ros/launch_ros/substitutions/find_package.py]: @ivanpauno can we `ensure_type_argument` on `package`?

Comment (ivanpauno) [reply launch_ros/launch_ros/substitutions/find_package.py]: We aren't usually doing that with, so I don't think that adding it here is a good idea.

**Comment by ivanpauno** [comment]
CI (build: --packages-up-to test_launch_ros, test: --packages-select launch_ros test_launch_ros, only fastrtps)
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/7140/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/3353/)
* macOS [Build Status](https://ci.ros2.org/job/ci_osx/5846/)
* Windows [Build Status](https://ci.ros2.org/job/ci_windows/6965/)

**Comment by ivanpauno** [comment]
Again windows CI:
* Windows [Build Status](https://ci.ros2.org/job/ci_windows/6981/)

**Comment by ivanpauno** [comment]
My test relied in that we weren't used `merge install` option. CI again after simple test:
* Windows [Build Status](https://ci.ros2.org/job/ci_windows/6982/)

File: launch_ros/launch_ros/substitutions/find_package.py (hunk 3 above)
Comment (hidmic) [reply launch_ros/launch_ros/substitutions/find_package.py]: Alright, fair enough. For now at least.

**Comment by hidmic** [comment]
> My test relied in that we weren't used merge install option. 

Hmm it'd be nice to at least use that package prefix to locate something that should be there. Otherwise, we are not testing anything.

**Comment by ivanpauno** [comment]
> > My test relied in that we weren't used merge install option.
> 
> Hmm it'd be nice to at least use that package prefix to locate something that should be there. Otherwise, we are not testing anything.

It was only testing that `ament_index_python.packages.get_package_prefix` found the package and didn't raise an exception. I pushed a better test in https://github.com/ros2/launch_ros/pull/22/commits/0208f444bae764e3d7c7d7f9683101f1cd47f075.

**Comment by ivanpauno** [comment]
CI (build: --packages-up-to test_launch_ros, test: --packages-select launch_ros test_launch_ros, only fastrtps)
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/7155/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/3367/)
* macOS [Build Status](https://ci.ros2.org/job/ci_osx/5859/)
* Windows [Build Status](https://ci.ros2.org/job/ci_windows/6984/)

**Pull request merged**